File: RealtySale.Shared/Data/Property.cs
﻿using System.ComponentModel.DataAnnotations.Schema;

namespace RealtySale.Shared.Data;

public class Property : BaseDataEntity
{
    public byte SellRent { get; set; }
    public string Name { get; set; } = string.Empty;
    public int PropertyTypeId { get; set; }
    public PropertyType? PropertyType { get; set; }
    public int Bhk { get; set; }
    public int FurnishingTypeId { get; set; }
    public FurnishingType? FurnishingType { get; set; }
    public int Price { get; set; }
    public int BuiltArea { get; set; }
    public int CarpetArea { get; set; }
    public string Address { get; set; } = string.Empty;
    public string Address2 { get; set; } = string.Empty;
    public int CityId { get; set; }
    public City? City { get; set; }
    public int FloorNo { get; set; }
    public int TotalFloors { get; set; }
    public bool ReadyToMove { get; set; }
    public string MainEntrance { get; set; } = string.Empty;
    public int Security { get; set; }
    public bool Gated { get; set; }
    public int Maintenance { get; set; }
    public DateTime EstPossessionOn { get; set; }
    public int Age { get; set; }
    public string Description { get; set; } = string.Empty;
    public ICollection<Photo>? Photos { get; set; }
    public DateTime PostedOn { get; set; } = DateTime.Now;
    [ForeignKey("User")]
    public int PostedBy { get; set; }
    public User? User { get; set; }
}

File: RealtySale.Client/src/app/models/Property.ts
import { IPhoto } from "./IPhoto.interface";
import { IPropertyBase } from "./IPropertyBase.interface";

export class Property implements IPropertyBase {
  id!: number;
  sellRent!: number;
  name!: string;
  propertyTypeId?: number;
  propertyType!: string;
  bhk!: number;
  furnishingTypeId?: number;
  furnishingType!: string;
  price!: number;
  builtArea!: number;
  carpetArea?: number;
  address!: string;
  address2?: string;
  cityId?: number;
  city!: string;
  floorNo?: string;
  totalFloors?: string;
  readyToMove!: boolean;
  age?: string;
  mainEntrance?: number;
  security?: number;
  gated?: boolean;
  maintenance?: number;
  image?: string;
  estPossessionOn?: string;
  description?: string;
  photos?: IPhoto[];
  postedOn?: string;
  postedBy!: number;
}
